src/amp-story-embed.js

```diff
@@ -150,7 +237,35 @@ self.onload = () => {
     const embed = embeds[i];
     const embedImpl = new AmpStoryEmbed(self, embed);
     embedImpl.buildCallback();
-    // TODO(Enriqe): add intersection observer that triggers layoutCallback().
-    embedImpl.layoutCallback();
+
+    if (IntersectionObserver && self === self.parent) {
```

Comment:
Nit: please split the code into different functions/methods, the deepest code line has 7 levels of indentation, making it super hard to follow